src/main/op/OP.ts

```diff
@@ -59,8 +65,22 @@ export class OP {
     requestJwtOrUri: string | URI,
     requestOpts?: { nonce?: string; verification?: InternalVerification | ExternalVerification }
   ): Promise<VerifiedAuthorizationRequest> {
-    const authorizationRequest = await AuthorizationRequest.fromUriOrJwt(requestJwtOrUri);
-    return await authorizationRequest.verify(this.newVerifyAuthorizationRequestOpts({ ...requestOpts }));
+    const authorizationRequest = await AuthorizationRequest.fromUriOrJwt(requestJwtOrUri).catch((error: Error) => {
+      opEventEmitter.emit(AuthorizationEvents.ON_AUTH_REQUEST_RECEIVED_FAILED, requestJwtOrUri, error);
```

Comment:
Could you change the order to be more similar to the one below.

So a then and then a catch, instead of a catch en then having the emit outside of the flow

Reply:
made some changes